scripts/ado-terraform-nagger.py

```diff
@@ -272,155 +399,333 @@ def extract_version(text, regex):
+
+        # Construct the working directory path
+        base_directory = os.getenv('BASE_DIRECTORY')
+        working_directory, components_list = create_working_dir_list(base_directory, system_default_working_directory, build_repo_suffix)
+        
+        output_warning = {
+            'terraform_version': {
+                'components': [],
+                'error_message': ''
+            },
+            'terraform_provider': {
+                'provider': {},
+                'error_message': ''
+            }
+        }
+
+        # for loop over dir componenets
+        for component in components_list:
+            print(f'COMPONENT: {component}')
+            full_path = f'{working_directory}{component}'
+            print(f'FULL PATH: {full_path}')
+
+            # Load deprecation map
+            config = load_file(args.filepath)
+
+            # Try to run `tfswitch' and 'terraform version --json` which is present in tf versions >= 0.13.0
+            command = ["tfswitch", "-b", terraform_binary_path]
+            run_command(command, full_path)
+
+            # try and do the terraform init 
+            command = ["terraform", "init", "-backend=false"]
+            output = run_command(command, full_path)
+            # check if tf has successfully init 
+            if not 'Terraform has been successfully initialized!' in output:
+                global errors_detected
+                relative_test_path = os.path.relpath(full_path, '../../../../../azp/_work/1/s')
+                logger.error(f'##vso[task.logissue type=error;]Repo structure invalid please see docs for further information: {relative_test_path}')
+                message = (f'Terraform initialized in an empty directory, Repo structure invalid please see docs for further information:\n{relative_test_path}')
+                errors_detected = True
+                log_message_slack(
                     slack_user_id,
                     slack_webhook_url,
-                    "warning",
-                    f"Provider {provider} is missing from version config. "
-                    "Please add it to the config in this file in order to "
-                    "compare it's versions.",
+                    message
                 )
+                raise SystemExit(1)
+
+            command = ["terraform", "version", "--json"]
+            result = json.loads(run_command(command, full_path))
+            terraform_version = result["terraform_version"]
+
+            # Append warning/error if flagged
+            warning, error_message = terraform_version_checker(terraform_version, config, current_date)
+
+            if warning == 'warning':
+                output_warning['terraform_version']['error_message'] = error_message
+                output_warning['terraform_version']['components'].append(component)
+            if warning == 'error':
+                add_error(output_warning, error_message, component)
```

Comment:
```suggestion
            alert_level, error_message = terraform_version_checker(terraform_version, config, current_date)

            if alert_level == 'warning':
                output_warning['terraform_version']['error_message'] = error_message
                output_warning['terraform_version']['components'].append(component)
            if alert_level == 'error':
                add_error(output_warning, error_message, component)
```

Perhaps? `if warning == 'error':` is a bit confusing after `if warning == 'warning'`

Reply:
this looks a lot better! was stuck on whether or not to change it but your suggestion is good 👍